modular/gater/admin_handler.go

```diff
@@ -377,6 +378,169 @@ func (g *GateModular) getChallengeInfoHandler(w http.ResponseWriter, r *http.Req
 	log.CtxDebug(reqCtx.Context(), "succeed to get challenge info")
 }
 
+// getChallengeInfoV2Handler handles get challenge piece info request. Current only greenfield
+// validator can challenge piece is store correctly. The challenge piece info includes:
```

Comment:
Currently, only greenfield validator can challenge piece whether is stored correctly.

Reply:
Fixed.